subsys/tracing/tracing_core.c

```diff
@@ -29,6 +29,8 @@
 #define TRACING_BACKEND_NAME "tracing_backend_posix"
 #elif defined CONFIG_TRACING_BACKEND_RAM
 #define TRACING_BACKEND_NAME "tracing_backend_ram"
+#elif defined CONFIG_TRACING_BACKEND_ADSP_MEMORY_WINDOW
+#define TRACING_BACKEND_NAME "tracing_backend_adsp_memory_window"
```

Comment:
This is... a really weird API.  Selecting a string value in a C file based on a bunch of kconfigs?  Why not just make it a string kconfig?  Or put a name field in the backend struct.  Not the fault of this PR obviously.